Src/Support/GoogleApis.Auth.PlatformServices_Shared/OAuth2/LocalServerCodeReceiver.cs

```diff
@@ -375,10 +378,33 @@ private HttpListener StartListener()
             return listener;
         }
 
-        private async Task<AuthorizationCodeResponseUrl> GetResponseFromListener(HttpListener listener)
+        private async Task<AuthorizationCodeResponseUrl> GetResponseFromListener(HttpListener listener, CancellationToken ct)
         {
+            // Set up cancellation. HttpListener.GetContextAsync() doesn't accept a cancellation token,
+            // the HttpListener needs to be stopped which immediately aborts the GetContextAsync() call.
+            var t = Task.Run(async () =>
+            {
+                try
+                {
+                    await Task.Delay(TimeSpan.FromHours(24), ct);
+                }
+                catch
+                {
+                    // Exception means the delay has been cancelled.
+                }
+                listener.Stop();
+            });
+
             // Wait to get the authorization code response.
-            var context = await listener.GetContextAsync().ConfigureAwait(false);
+            HttpListenerContext context;
+            try
+            {
+                context = await listener.GetContextAsync().ConfigureAwait(false);
+            }
+            catch (HttpListenerException e) when (e.ErrorCode == 995) // 995 = ERROR_OPERATION_ABORTED
```

Comment:
Or when `ct.IsCancellationRequested`, and just call `ct.ThrowIfCancellationRequested()`?